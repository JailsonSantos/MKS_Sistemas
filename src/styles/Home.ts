import styled from "styled-components";
import { mobile, tablet } from "./responsive";

export const Container = styled.div`
  flex: 1;
`;

export const Main = styled.main`
  flex: 1;
  gap: 20px;
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  overflow: hidden;
  margin: 40px auto;
  max-width: 1024px;
  padding-left: 50px;

  ${tablet({
  width: '100%',
  paddingLeft: '50px',
  justifyContent: 'flex-start'
})};
  
  ${mobile({
  width: '100%',
  flexWrap: 'nowrap',
  alignItems: 'center',
  padding: '0',
  flexDirection: 'column',
})};

`;
