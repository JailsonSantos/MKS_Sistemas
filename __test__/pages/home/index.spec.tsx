import Home from '@/pages';
import '@testing-library/jest-dom';
import { screen } from '@testing-library/dom';
import { render } from '@testing-library/react';
import { ProductsFake } from '@/data/ProductsFake';

jest.mock("react-redux", () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

describe('Home component', () => {
  it("Shoud render component", () => {

    const { container } = render(<Home products={ProductsFake} />)

    expect(container).toMatchSnapshot()

  });

  it('List exits in component', () => {

    render(<Home products={ProductsFake} />)

    expect(screen.queryAllByText((content, element) => {
      return element?.tagName.toLowerCase() === 'span' && content.startsWith('Headset Cloud Stinger')
    }));

    expect(screen.queryAllByText((content, element) => {
      return element?.tagName.toLowerCase() === 'span' && content.startsWith('Headset Cloud Revolver')
    }));

    expect(screen.queryAllByText((content, element) => {
      return element?.tagName.toLowerCase() === 'span' && content.startsWith('iPad')
    }));

    expect(screen.queryAllByText((content, element) => {
      return element?.tagName.toLowerCase() === 'span' && content.startsWith('Apple Watch Series 7')
    }));

    expect(screen.queryAllByText((content, element) => {
      return element?.tagName.toLowerCase() === 'span' && content.startsWith('iPhone 12 64 GB')
    }));

    expect(screen.queryAllByText((content, element) => {
      return element?.tagName.toLowerCase() === 'span' && content.startsWith('Macbook Air')
    }));

    expect(screen.queryAllByText((content, element) => {
      return element?.tagName.toLowerCase() === 'span' && content.startsWith('AirPods')
    }));

    expect(screen.queryAllByText((content, element) => {
      return element?.tagName.toLowerCase() === 'span' && content.startsWith('Iphone 11 128 GB')
    }));

  });
});