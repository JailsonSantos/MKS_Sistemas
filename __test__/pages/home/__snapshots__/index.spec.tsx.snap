// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Shoud render component 1`] = `
<div>
  <div
    class="sc-ezOQGI cVdiFe"
  >
    <div
      class="sc-dIfARi dxFhTW"
    >
      <div>
        <div
          class="sc-hHTYSt dRHmMK"
        >
          MKS 
          <span>
            Sistemas
          </span>
        </div>
      </div>
      <button
        aria-controls="radix-:r3:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="sc-fnGiBr eXTSQQ"
        data-state="closed"
        type="button"
      >
        <svg
          fill="currentColor"
          height="24"
          viewBox="0 0 256 256"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <path
            d="M184,184H69.8L41.9,30.6A8,8,0,0,0,34.1,24H16"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="24"
          />
          <circle
            cx="80"
            cy="204"
            fill="none"
            r="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="24"
          />
          <circle
            cx="184"
            cy="204"
            fill="none"
            r="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="24"
          />
          <path
            d="M62.5,144H188.1a15.9,15.9,0,0,0,15.7-13.1L216,64H48"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="24"
          />
        </svg>
        <span />
      </button>
    </div>
    <main
      class="sc-bYMpWt hZjaGm"
    >
      <div
        class="sc-kgTSHT kShjZj"
      >
        <div
          class="sc-fLcnxK fGfCEs"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="--animation-duration: 4s; width: 111px; height: 138px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              style="margin-left: 0.2rem;"
            >
              <span
                class="react-loading-skeleton"
                style="--animation-duration: 4s; width: 100px; height: 30px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
              >
                ‌
              </span>
            </span>
            <span
              style="margin-left: 0.2rem;"
            >
              <span
                class="react-loading-skeleton"
                style="--animation-duration: 4s; width: 100px; height: 30px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
              >
                ‌
              </span>
            </span>
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="--animation-duration: 4s; width: 210px; height: 50px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton"
            style="border-top-left-radius: 0; border-top-right-radius: 0; border-bottom-left-radius: 0.5rem; border-bottom-right-radius: 0.5rem; position: relative; bottom: -10px; --animation-duration: 4s; width: 220px; height: 35px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
          >
            ‌
          </span>
          <br />
        </span>
      </div>
      <div
        class="sc-kgTSHT kShjZj"
      >
        <div
          class="sc-fLcnxK fGfCEs"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="--animation-duration: 4s; width: 111px; height: 138px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              style="margin-left: 0.2rem;"
            >
              <span
                class="react-loading-skeleton"
                style="--animation-duration: 4s; width: 100px; height: 30px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
              >
                ‌
              </span>
            </span>
            <span
              style="margin-left: 0.2rem;"
            >
              <span
                class="react-loading-skeleton"
                style="--animation-duration: 4s; width: 100px; height: 30px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
              >
                ‌
              </span>
            </span>
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="--animation-duration: 4s; width: 210px; height: 50px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton"
            style="border-top-left-radius: 0; border-top-right-radius: 0; border-bottom-left-radius: 0.5rem; border-bottom-right-radius: 0.5rem; position: relative; bottom: -10px; --animation-duration: 4s; width: 220px; height: 35px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
          >
            ‌
          </span>
          <br />
        </span>
      </div>
      <div
        class="sc-kgTSHT kShjZj"
      >
        <div
          class="sc-fLcnxK fGfCEs"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="--animation-duration: 4s; width: 111px; height: 138px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              style="margin-left: 0.2rem;"
            >
              <span
                class="react-loading-skeleton"
                style="--animation-duration: 4s; width: 100px; height: 30px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
              >
                ‌
              </span>
            </span>
            <span
              style="margin-left: 0.2rem;"
            >
              <span
                class="react-loading-skeleton"
                style="--animation-duration: 4s; width: 100px; height: 30px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
              >
                ‌
              </span>
            </span>
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="--animation-duration: 4s; width: 210px; height: 50px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton"
            style="border-top-left-radius: 0; border-top-right-radius: 0; border-bottom-left-radius: 0.5rem; border-bottom-right-radius: 0.5rem; position: relative; bottom: -10px; --animation-duration: 4s; width: 220px; height: 35px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
          >
            ‌
          </span>
          <br />
        </span>
      </div>
      <div
        class="sc-kgTSHT kShjZj"
      >
        <div
          class="sc-fLcnxK fGfCEs"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="--animation-duration: 4s; width: 111px; height: 138px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              style="margin-left: 0.2rem;"
            >
              <span
                class="react-loading-skeleton"
                style="--animation-duration: 4s; width: 100px; height: 30px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
              >
                ‌
              </span>
            </span>
            <span
              style="margin-left: 0.2rem;"
            >
              <span
                class="react-loading-skeleton"
                style="--animation-duration: 4s; width: 100px; height: 30px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
              >
                ‌
              </span>
            </span>
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="--animation-duration: 4s; width: 210px; height: 50px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton"
            style="border-top-left-radius: 0; border-top-right-radius: 0; border-bottom-left-radius: 0.5rem; border-bottom-right-radius: 0.5rem; position: relative; bottom: -10px; --animation-duration: 4s; width: 220px; height: 35px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
          >
            ‌
          </span>
          <br />
        </span>
      </div>
      <div
        class="sc-kgTSHT kShjZj"
      >
        <div
          class="sc-fLcnxK fGfCEs"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="--animation-duration: 4s; width: 111px; height: 138px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              style="margin-left: 0.2rem;"
            >
              <span
                class="react-loading-skeleton"
                style="--animation-duration: 4s; width: 100px; height: 30px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
              >
                ‌
              </span>
            </span>
            <span
              style="margin-left: 0.2rem;"
            >
              <span
                class="react-loading-skeleton"
                style="--animation-duration: 4s; width: 100px; height: 30px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
              >
                ‌
              </span>
            </span>
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="--animation-duration: 4s; width: 210px; height: 50px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton"
            style="border-top-left-radius: 0; border-top-right-radius: 0; border-bottom-left-radius: 0.5rem; border-bottom-right-radius: 0.5rem; position: relative; bottom: -10px; --animation-duration: 4s; width: 220px; height: 35px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
          >
            ‌
          </span>
          <br />
        </span>
      </div>
      <div
        class="sc-kgTSHT kShjZj"
      >
        <div
          class="sc-fLcnxK fGfCEs"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="--animation-duration: 4s; width: 111px; height: 138px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              style="margin-left: 0.2rem;"
            >
              <span
                class="react-loading-skeleton"
                style="--animation-duration: 4s; width: 100px; height: 30px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
              >
                ‌
              </span>
            </span>
            <span
              style="margin-left: 0.2rem;"
            >
              <span
                class="react-loading-skeleton"
                style="--animation-duration: 4s; width: 100px; height: 30px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
              >
                ‌
              </span>
            </span>
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="--animation-duration: 4s; width: 210px; height: 50px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton"
            style="border-top-left-radius: 0; border-top-right-radius: 0; border-bottom-left-radius: 0.5rem; border-bottom-right-radius: 0.5rem; position: relative; bottom: -10px; --animation-duration: 4s; width: 220px; height: 35px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
          >
            ‌
          </span>
          <br />
        </span>
      </div>
      <div
        class="sc-kgTSHT kShjZj"
      >
        <div
          class="sc-fLcnxK fGfCEs"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="--animation-duration: 4s; width: 111px; height: 138px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              style="margin-left: 0.2rem;"
            >
              <span
                class="react-loading-skeleton"
                style="--animation-duration: 4s; width: 100px; height: 30px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
              >
                ‌
              </span>
            </span>
            <span
              style="margin-left: 0.2rem;"
            >
              <span
                class="react-loading-skeleton"
                style="--animation-duration: 4s; width: 100px; height: 30px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
              >
                ‌
              </span>
            </span>
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="--animation-duration: 4s; width: 210px; height: 50px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton"
            style="border-top-left-radius: 0; border-top-right-radius: 0; border-bottom-left-radius: 0.5rem; border-bottom-right-radius: 0.5rem; position: relative; bottom: -10px; --animation-duration: 4s; width: 220px; height: 35px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
          >
            ‌
          </span>
          <br />
        </span>
      </div>
      <div
        class="sc-kgTSHT kShjZj"
      >
        <div
          class="sc-fLcnxK fGfCEs"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="--animation-duration: 4s; width: 111px; height: 138px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              style="margin-left: 0.2rem;"
            >
              <span
                class="react-loading-skeleton"
                style="--animation-duration: 4s; width: 100px; height: 30px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
              >
                ‌
              </span>
            </span>
            <span
              style="margin-left: 0.2rem;"
            >
              <span
                class="react-loading-skeleton"
                style="--animation-duration: 4s; width: 100px; height: 30px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
              >
                ‌
              </span>
            </span>
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton"
              style="--animation-duration: 4s; width: 210px; height: 50px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton"
            style="border-top-left-radius: 0; border-top-right-radius: 0; border-bottom-left-radius: 0.5rem; border-bottom-right-radius: 0.5rem; position: relative; bottom: -10px; --animation-duration: 4s; width: 220px; height: 35px; border-radius: 0.5rem; --base-color: rgba(238,238,238, 0.5); --highlight-color: rgba(0,128,178, 0.1);"
          >
            ‌
          </span>
          <br />
        </span>
      </div>
    </main>
    <div
      class="sc-bcXHqe encDOS"
    >
      <span>
        MKS sistemas © Todos os direitos reservados.
      </span>
    </div>
  </div>
</div>
`;
